test/internal/layers/lazy.go

```diff
@@ -0,0 +1,165 @@
+//go:build windows
+
+package layers
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"sync"
+	"testing"
+
+	"github.com/Microsoft/go-winio"
+	"github.com/google/go-containerregistry/pkg/crane"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+
+	"github.com/Microsoft/hcsshim/ext4/tar2ext4"
+	"github.com/Microsoft/hcsshim/internal/log"
+	"github.com/Microsoft/hcsshim/internal/security"
+	"github.com/Microsoft/hcsshim/internal/wclayer"
+	"github.com/Microsoft/hcsshim/pkg/ociwclayer"
+
+	"github.com/Microsoft/hcsshim/test/internal/constants"
+	"github.com/Microsoft/hcsshim/test/internal/util"
+)
+
+// helper utilities for dealing with images
+
+type LazyImageLayers struct {
+	Image    string
+	Platform string
+	// TempPath is the path to create a temporary directory in.
+	// Defaults to [os.TempDir] if left empty.
+	TempPath string
+	// dedicated directory, under [TempPath], to store layers in
+	dir    string
+	once   sync.Once
+	layers []string
+}
+
+// Close removes the downloaded image layers.
+//
+// Does not take a [testing.TB] so it can be used in TestMain or init.
+func (x *LazyImageLayers) Close(ctx context.Context) error {
```

Comment:
Does this only remove windows image layers?

Reply:
nope, `DestroyLayer` removes the entire directory, regardless of whether its a WCOW layer or not

Reply:
Okay, could we add a comment for that?